sources/packages/backend/apps/api/src/route-controllers/designation-agreement/_tests_/e2e/designation-agreement.institutions.controller.submitDesignationAgreement.e2e-spec.ts

```diff
@@ -172,6 +173,86 @@ describe("DesignationAgreementInstitutionsController(e2e)-submitDesignationAgree
       .expect(HttpStatus.UNPROCESSABLE_ENTITY);
   });
 
+  it("Should return an unprocessable entity when no location is selected for designation.", async () => {
+    // Arrange
+    const payload = {
+      dynamicData: {
+        eligibilityOfficers: [],
+        enrolmentOfficers: [],
+        scheduleA: false,
+        scheduleB: false,
+        scheduleD: false,
+        legalAuthorityName: "SIMS COLLF",
+        legalAuthorityEmailAddress: "test@gov.bc.ca",
+        agreementAccepted: false,
+      },
+      locations: [
+        {
+          locationId: collegeFLocation.id,
+          requestForDesignation: false,
+        },
+      ],
+    };
+    const institutionUserToken = await getInstitutionToken(
+      InstitutionTokenTypes.CollegeFAdminLegalSigningUser,
+    );
+    const endpoint = "/institutions/designation-agreement";
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .post(endpoint)
+      .send(payload)
+      .auth(institutionUserToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.UNPROCESSABLE_ENTITY)
+      .expect((response) => {
+        expect(response.body.errorType).toBe(
+          NO_LOCATION_SELECTED_FOR_DESIGNATION,
+        );
+      });
+  });
+
+  it("Should return an unprocessable entity when all locations have requestForDesignation set to false.", async () => {
+    // Arrange
+    const payload = {
+      dynamicData: {
+        eligibilityOfficers: [],
+        enrolmentOfficers: [],
+        scheduleA: false,
+        scheduleB: false,
+        scheduleD: false,
+        legalAuthorityName: "SIMS COLLF",
+        legalAuthorityEmailAddress: "test@gov.bc.ca",
+        agreementAccepted: false,
+      },
+      locations: [
+        {
+          locationId: collegeFLocation.id,
+          requestForDesignation: false,
+        },
+        {
+          locationId: collegeFLocation.id,
+          requestForDesignation: false,
+        },
+      ],
+    };
+    const institutionUserToken = await getInstitutionToken(
+      InstitutionTokenTypes.CollegeFAdminLegalSigningUser,
+    );
+    const endpoint = "/institutions/designation-agreement";
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .post(endpoint)
+      .send(payload)
+      .auth(institutionUserToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.UNPROCESSABLE_ENTITY)
```

Comment:
The preferred project way of executing these assertions would be similar to the one below.
```ts
.expect(HttpStatus.UNPROCESSABLE_ENTITY)
.expect({
  message:
    "Cannot remove a bypass when application restriction bypass is not active.",
  errorType: APPLICATION_RESTRICTION_BYPASS_IS_NOT_ACTIVE,
});
```